app/src/main/java/unipd/se18/ocrcamera/Inci.java

```diff
@@ -0,0 +1,148 @@
+     */
+    private void loadDB(InputStream inputStream){
+        //define 0 the name of the ingredients
+        int name=0;
+        //define 1 the description of the ingredients
+        int description=1;
+        //open Buffer Reader for read the database
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream, Charset.forName("UTF-8")));
+        String ingredient;
+        try {
+            //slide every line of the database and add in the appropriate list
+            while ((ingredient = reader.readLine()) != null) {
+                String[] inci=ingredient.split(";");
+                listInci.add(inci[name].trim());
+                //TODO listInciDescription
+                //listInciDescription.add(inci[description].trim());
+            }
+            Log.d("inci", "create listInci");
+        }catch (Exception e){
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * @author Giovanni Fasan(g1)
+     * @return String with ingredients list of the label
+     * Concatenate all the ingredients found in a string. It puts the ingredidients in bold
+     */
+
+    public String ingredientsFoundToString() {
+        String inci="";
+        for (int j=0; j<this.listIngredientFound.size(); j++){
+            inci = inci + "<b>" + listIngredientFound.get(j) + "</b>; ";
+        }
+
+        return inci;
+    }
+
+    /**
+     * @author Giovanni Fasan(g1)
+     * @param text String in which you have to find the ingredients
+     * loads all the ingredients inci found in the ArrayList listIngredientFound
+     */
+    private void findIngredientsList_old(String text) {
+        listIngredientFound = new ArrayList<String>();
+        //remove every \n
+        //split by Ingredients to reduce the number of words to be analyzed
+        String noEnd = text.replaceAll("\n", " ");
+        String [] word = noEnd.split(",");
+        for (int i = 0; i < word.length; i++) {
+            for (int j=0; j<this.listInci.size(); j++){
+                if (word[i].trim().toUpperCase().equals(listInci.get(j))) {
+                    listIngredientFound.add(listInci.get(j));
+                }
+            }
+        }
+
+    }
+
+
+    /**
+     * @author Giovanni Fasan(g1)
+     * @param text String in which you have to find the ingredients
+     * loads all the ingredients inci found in the ArrayList listIngredientFound
+     */
+    public void findIngredientsList(String text) {
+        listIngredientFound = new ArrayList<String>();
+        //remove every \n and every :
+        //split by Ingredients to reduce the number of words to be analyzed
+        String[] noEnd = text.replaceAll("\n", " ").replaceAll(":", " ").split("Ingredients");
+        if (noEnd.length==2){
+            //if I have a length of the vector is 2, I take the second part of the vector, that is the one where there is the list of ingredients
+            String [] word = noEnd[1].split(",");
+            Log.d("inci", "=2");
```

Comment:
Could a good idea be using a final TAG String instead the tag of the class hard coded here?

Reply:
I forgot to remove it